testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestExternalAuthorizerOAuth2.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.jdbc;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import io.trino.jdbc.TrinoDriverUri;
+import io.trino.tempto.ProductTest;
+import io.trino.tests.TpchTableResults;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tempto.query.QueryResult.forResultSet;
+import static io.trino.tests.TestGroups.OAUTH2;
+import static org.openqa.selenium.support.ui.ExpectedConditions.elementToBeClickable;
+
+public class TestExternalAuthorizerOAuth2
+        extends ProductTest
+{
+    @Inject
+    @Named("databases.presto.jdbc_url")
+    String jdbcUrl;
+
+    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    @Test(groups = OAUTH2)
+    public void shouldAuthenticateAndExecuteQuery()
+            throws SQLException
+    {
+        prepareSeleniumHandler();
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, properties);
+                PreparedStatement statement = connection.prepareStatement("select * from tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThat(forResultSet(results)).matches(TpchTableResults.PRESTO_NATION_RESULT);
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void clean()
+    {
+        executor.shutdown();
+    }
+
+    private void prepareSeleniumHandler()
+    {
+        TrinoDriverUri.setRedirectHandler(uri -> executor.execute(() -> {
```

Comment:
Don't you need to capture future, to see no exceptions were thrown and if so to capture the stacktrace of it?